scaaml/io/dataset.py

```diff
@@ -289,8 +291,9 @@ def new_shard(self, key: list, part: int, group: int, split: str,
         if split not in Dataset.SPLITS:
             raise ValueError(f"Invalid split, must be in {Dataset.SPLITS}")
 
-        if part < 0 or part > 10:
-            raise ValueError("Invalid part value -- muse be in [0, 10]")
+        if part < 0 or part > self.MAX_PART_NUMBER:
+            raise ValueError(
+                "Invalid part value -- must be in [0, self.MAX_PART_NUMBER]")
```

Comment:
Shouldn't that be an f-string with `self.MAX_PART_NUMBER` surrounded by curly braces?